### ISSUE TITLE
power_aix_bootstrap should use ansible_host, ansible_password, etc.

### ISSUE THREAD

**Issue by stephenulmer** [IBM/ansible-power-aix#106]
_NB: I really thought I already opened an issue for this the other day, but it doesn't seem to exist (open or closed). I apologize if this is a duplicate!_

**Is your feature request related to a problem? Please describe.**
The power_aix_bootstrap role has to copy file to the target in order to install YUM and it's dependencies. This can cause the play to stop and wait for password entry, or to fail if the target is not resolvable. Ansible already handles these situations internally, but does not provide an obvious way to copy a file to a target without having Python installed on the target.

**Describe the solution you'd like**
The role should use a copy equivalent of the **raw** module, so that it would automatically have the same behavior as the rest of Ansible. Ansible does not provide such an action, so one is implemented in PR #104.

**Describe alternatives you've considered**
I looked at implementing all (well, most) of the ansible_* variables I need in the tasks of the role, but it got out of hand very quickly. The result would not be readable or easily maintained.

**Additional context**
I'm building clusters of AIX systems from bare metal up using Ansible, and it's not feasible to look up and type in 64 randomly generated root passwords every time we re-build. Also, the machines work for us. :)

**Comment by pvtorres** [comment]
Hi Stephen, 
I recreated the problem and it seems the yum_install.yml in the power_aix_bootstrap  role is the one causing the problem. 
The yum_install.yml tries to do an scp command to the target that is being handled during the execution of the role. 
But it is using the inventory_hostname variable which is basically replaced directly with the "host" without expanding what the host is.  I replaced the direct use of the inventory_hostname with the next:
"{{ hostvars[inventory_hostname].inventory_hostname }}"
This seems to do the trick and it will be replaced with the expanded hostname (IP address, short or long hostname according to the inventory definition) it also does not requires the ansible_facts enabled.

**Comment by stephenulmer** [comment]
Yes, my pull request patches that very file (for those reasons).

That was a great way to resolve the name resolution problem! I had just used the existing connection method and let Ansible resolve it for me, but it required python to get that low in the stack…

Unfortunately, when running scp in a task I still can’t find a good way to let Ansible use its configuration information about the host to authenticate. So when deploying a cluster, this asks for login information about each node… the reason I wrote the module was so that Ansible could try other ssh keys and any passwords configured with ansible_*_password, as well as honor variables that modify the ssh parameters for the connection (like changing the port, cipher type, which identity to use, maybe don’t check the host keys during the bootstrap).

I don’t think it’s reasonable to make the end-user of an automation product enter passwords manually when the product they’re using will do it for them.

-- 
Stephen
